server/controllers/finance/patientInvoice.js

```diff
@@ -211,11 +211,23 @@ function create(req, res, next) {
   // make sure that the dates have been properly transformed before insert
   if (invoice.date) {
     invoice.date = new Date(invoice.date);
+  }else{
+    throw new BadRequest(`An invoice is not allowed to miss a date.`);
+  }
+
+  if(items.length < 1){
+    throw new BadRequest(`An invoice is not allowed to miss a items.`);
+  }
+
+  if(!invoice.description){
```

Comment:
Does the database not enforce these?  I thought it returned a 400 request already...